player/mcts/mcts_node.py

```diff
@@ -75,13 +76,15 @@ def build_tree(self, should_continue):
             # Backpropagate
             while node:
                 node._number_of_runs += 1
-                node._number_of_wins = state.get_results_for_player(self.game_state.is_first_player_move)
+                current_reward = state.get_results_for_player(self.game_state.is_first_player_move)
+                node._reward += current_reward
+                node._reward_squared += current_reward ** 2
                 node = node.parent
 
             iteration += 1
 
     def select_child(self):
-        return max(self.children, key=lambda child: child.uct)
+        return max(self.children, key=lambda child: child.eval)
 
     def add_child(self, action: int, game_state: IGameState):
         child = MctsNode(game_state, action, self)
```

Comment:
I guess if we want to use other implementations for mcts_node we should also create instance of proper type here as well